Purchases/Caching/RCDeviceCache.m

```diff
@@ -62,27 +66,38 @@ - (void)clearCachesForAppUserID:(NSString *)appUserID
 
 - (BOOL)isCacheStale {
     NSTimeInterval timeSinceLastCheck = -[self.cachesLastUpdated timeIntervalSinceNow];
-    return !(self.cachesLastUpdated != nil && timeSinceLastCheck < 60. * 5);
+    return !(self.cachesLastUpdated != nil && timeSinceLastCheck < CACHE_DURATION_IN_SECONDS);
+}
+
+- (BOOL)isOfferingsCacheStale {
+    return self.offeringsCachedObject.isCacheStale;
+}
+
+- (RCOfferings * _Nullable)cachedOfferings {
+    return self.offeringsCachedObject.cachedInstance;
 }
 
 - (void)resetCachesTimestamp
 {
-    self.cachesLastUpdated = [NSDate date];
+    NSDate *now = [NSDate date];
+    self.cachesLastUpdated = now;
+    [self.offeringsCachedObject updateCacheTimestampWithDate:now];
 }
 
 - (void)clearCachesTimestamp
 {
     self.cachesLastUpdated = nil;
+    [self.offeringsCachedObject clearCacheTimestamp];
 }
 
 - (void)cacheOfferings:(RCOfferings *)offerings
 {
-    self.cachedOfferings = offerings;
+    [self.offeringsCachedObject cacheInstance:offerings date:[NSDate date]];
```

Comment:
`- (void)cachePurchaserInfo:(NSData *)data forAppUserID:(NSString *)appUserID` is not setting the `cachesLastUpdated` but `- (void)cacheOfferings:(RCOfferings *)offerings` does. 

I think we have a bug where the cached PurchaserInfo is updated but not the cachesLastUpdated. I see a couple places where this is happening (in restore and in handleReceiptPostWithTransaction)

Reply:
I actually thought that it might be a bug. I didn't want to be overly aggressive in modifying behavior, though, because I wasn't sure whether that was intended behavior.
My go-to for this kind of thing would be to have the method never allow you to set the cached object without setting the date, so that you physically can't make the mistake.

Reply:
b21aa1cc4184feab1ce305030cd44d7acb455585

Reply:
I guess that having it separate is meant for thread safety? i.e.: so that nobody attempts to refresh again while we're fetching new data?

Reply:
one defensive approach would be to set the timestamp _before_ fetching so that nobody else tries a fetch, and also update it in the method so that you can't forget to set it. thoughts?

Reply:
Can we just wrap both calls with a @synchronized  

https://github.com/RevenueCat/purchases-ios/pull/168/files#r373752476